Panic on acs-engine 0.14.6  generate when adding "enableAggregatedAPIs" to propeties

<!-- Thanks for filing an issue! Before hitting the button, please answer these questions. It's helpful to search the existing GitHub issues first. It's likely that another user has already reported the issue you're facing, or it's a known issue that we're already aware of-->

**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): ISSUE

---

**What version of acs-engine?**:

Release linux AMD 64 - acs-engine-v0.14.6
https://github.com/Azure/acs-engine/releases/download/v0.14.6/acs-engine-v0.14.6-linux-amd64.tar.gz

---


<!--
If this is a ISSUE, please:
  - Fill in as much of the template below as you can.  If you leave out
    information, we can't help you as well.

If this is a FEATURE REQUEST, please:
  - Describe *in detail* the feature/behavior/change you'd like to see.

In both cases, be ready for followup questions, and please respond in a timely
manner.  If we can't reproduce a bug or think a feature already exists, we
might close your issue.  If we're wrong, PLEASE feel free to reopen it and
explain why.
-->

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
Using ubuntu bash on windows 10, but almost certain that it would be reproduceable in other shells
Downloaded acs engine like this:

```bash
curl -L https://github.com/Azure/acs-engine/releases/download/v0.14.6/acs-engine-v0.14.6-linux-amd64.tar.gz | tar zx
cp acs-engine-v0.14.6-linux-amd64/acs-engine .
chmod +x acs-engine
```

Used the following json file:

```json
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.10",
      "kubernetesConfig": {
          "enableAggregatedAPIs": true
      }

    },
    "masterProfile": {
      "count": 3,
      "dnsPrefix": "wink8s001cluster",
      "vmSize": "Standard_DS2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "winagents",
        "count": 3,
        "vmSize": "Standard_D2s_v3",
        "availabilityProfile": "AvailabilitySet",
        "osType": "Windows"
      }
    ],
    "windowsProfile": {
      "adminUsername": "azureuser",
      "adminPassword": "4p4ssw0rd!"
    },
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "",
      "secret": ""
    }
  }
}
```

You run
```
./acs-engine generate cluster.json
```

And you get the following error:

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6152f6]

goroutine 1 [running]:
github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver.(*Version).Prerelease(...)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver/version.go:146
github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver.constraintLessThan(0x0, 0xc420494980, 0x1)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver/constraints.go:256 +0x26
github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver.(*constraint).check(0xc420494980, 0x0, 0xc420734d80)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver/constraints.go:159 +0x38
github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver.Constraints.Check(0xc4206fc5a0, 0x1, 0x1, 0x0, 0x0)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/Masterminds/semver/constraints.go:49 +0x75
github.com/Azure/acs-engine/pkg/api/vlabs.(*OrchestratorProfile).Validate(0xc420050d40, 0x17bae00, 0xc42073af00, 0x0)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/pkg/api/vlabs/validate.go:94 +0x830
github.com/Azure/acs-engine/pkg/api/vlabs.(*Properties).Validate(0xc42073af00, 0x600, 0x8d8, 0xc420795960)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/pkg/api/vlabs/validate.go:310 +0xe3
github.com/Azure/acs-engine/pkg/api.(*Apiloader).LoadContainerService(0xc420795bc0, 0xc4200da900, 0x6d8, 0x8d8, 0xc4206bb1a8, 0x5, 0x1, 0x0, 0x19d9488, 0xc420795a20, ...)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/pkg/api/apiloader.go:170 +0x6f4
github.com/Azure/acs-engine/pkg/api.(*Apiloader).DeserializeContainerService(0xc420795bc0, 0xc4200da900, 0x6d8, 0x8d8, 0x1, 0x0, 0x0, 0x7fe233c30000, 0x0, 0x4979a9, ...)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/pkg/api/apiloader.go:44 +0x150
github.com/Azure/acs-engine/pkg/api.(*Apiloader).LoadContainerServiceFromFile(0xc420795bc0, 0x7fffe2700c25, 0xe, 0xc420770001, 0x0, 0x0, 0xc420778660, 0x19437ba, 0x5, 0xc4206eb5a8)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/pkg/api/apiloader.go:33 +0x211
github.com/Azure/acs-engine/cmd.(*generateCmd).validate(0xc4203af490, 0xc420774b40, 0xc420734600, 0x1, 0x1, 0x19429ce, 0x4)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/cmd/generate.go:98 +0x182
github.com/Azure/acs-engine/cmd.newGenerateCmd.func1(0xc420774b40, 0xc420734600, 0x1, 0x1, 0x0, 0x0)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/cmd/generate.go:48 +0x61
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).execute(0xc420774b40, 0xc4207345d0, 0x1, 0x1, 0xc420774b40, 0xc4207345d0)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:647 +0x3f1
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420774480, 0xc4207758c0, 0xc420775680, 0xc420775200)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:726 +0x2fe
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420774480, 0xc420084008, 0x0)
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:685 +0x2b
main.main()
        /Users/jackfrancis/work/src/github.com/Azure/acs-engine/main.go:12 +0x74
```

**What you expected to happen**:

Give me a proper error message that it doesn't support  `enableAggregatedAPIs`, and not list @jackfrancis directories.

**How to reproduce it** (as minimally and precisely as possible):

Follow steps above.

**Anything else we need to know**:

--- a/pkg/api/vlabs/validate.go
+++ b/pkg/api/vlabs/validate.go
@@ -89,11 +89,18 @@ func (o *OrchestratorProfile) Validate(isUpdate bool) error {
 					return err
 				}
 				if o.KubernetesConfig.EnableAggregatedAPIs {
-					sv, _ := semver.NewVersion(o.OrchestratorVersion)
-					cons, _ := semver.NewConstraint("<" + "1.7.0")
+					sv, err := semver.NewVersion(version)
+					if err != nil {
+						return fmt.Errorf("could not validate version %s", version)
+					}
+					minVersion := "1.7.0"
+					cons, err := semver.NewConstraint("<" + minVersion)
+					if err != nil {
+						return fmt.Errorf("could not apply semver constraint < %s against version %s", minVersion, version)
+					}
 					if cons.Check(sv) {
 						return fmt.Errorf("enableAggregatedAPIs is only available in Kubernetes version %s or greater; unable to validate for Kubernetes version %s",
-							"1.7.0", o.OrchestratorVersion)
+							minVersion, version)
 					}
 
 					if o.KubernetesConfig.EnableRbac != nil {
@@ -103,11 +110,17 @@ func (o *OrchestratorProfile) Validate(isUpdate bool) error {
 					}
 
 					if helpers.IsTrueBoolPointer(o.KubernetesConfig.EnableDataEncryptionAtRest) {
-						sv, _ := semver.NewVersion(o.OrchestratorVersion)
-						cons, _ := semver.NewConstraint("<" + "1.7.0")
+						sv, err := semver.NewVersion(version)
+						if err != nil {
+							return fmt.Errorf("could not validate version %s", version)
+						}
+						cons, err := semver.NewConstraint("<" + minVersion)
+						if err != nil {
+							return fmt.Errorf("could not apply semver constraint < %s against version %s", minVersion, version)
+						}
 						if cons.Check(sv) {
 							return fmt.Errorf("enableDataEncryptionAtRest is only available in Kubernetes version %s or greater; unable to validate for Kubernetes version %s",
-								"1.7.0", o.OrchestratorVersion)
+								minVersion, o.OrchestratorVersion)
 						}
 					}
 				}
@@ -115,11 +128,18 @@ func (o *OrchestratorProfile) Validate(isUpdate bool) error {
 					if !helpers.IsTrueBoolPointer(o.KubernetesConfig.EnableRbac) {
 						return fmt.Errorf("enablePodSecurityPolicy requires the enableRbac feature as a prerequisite")
 					}
-					sv, _ := semver.NewVersion(o.OrchestratorVersion)
-					cons, _ := semver.NewConstraint("<" + "1.8.0")
+					sv, err := semver.NewVersion(version)
+					if err != nil {
+						return fmt.Errorf("could not validate version %s", version)
+					}
+					minVersion := "1.8.0"
+					cons, err := semver.NewConstraint("<" + minVersion)
+					if err != nil {
+						return fmt.Errorf("could not apply semver constraint < %s against version %s", minVersion, version)
+					}
 					if cons.Check(sv) {
 						return fmt.Errorf("enablePodSecurityPolicy is only supported in acs-engine for Kubernetes version %s or greater; unable to validate for Kubernetes version %s",
-							"1.8.0", o.OrchestratorVersion)
+							minVersion, version)
 					}
 				}
 			}

